### DISCUSSION TITLE
Kafka Liveness probe not working for some cases

### DISCUSSION THREAD

**Discussion by vshinde727** [strimzi/strimzi-kafka-operator#10332] [General]
HI - Starting this discussion to get some views on current liveness probe in strimzi. We are facing one problem in some cases, when kafka pod gets into ready state but somehow does not respond to requests on any port.

Kafka - 3.2.0
Strimzi - 0.29

`
[kafka@kafka-cluster-kafka-0 kafka]$ netstat -antp | grep LIST
tcp6       0      0 :::9404                 :::*                    LISTEN      143/java
tcp6      51      0 :::9091                 :::*                    LISTEN      143/java
tcp6       0      0 :::9090                 :::*                    LISTEN      143/java
tcp6      51      0 :::9093                 :::*                    LISTEN      143/java
tcp6      51      0 :::9092                 :::*                    LISTEN      143/java
tcp6       0      0 :::42095                :::*                    LISTEN      143/java

[kafka@kafka-cluster-kafka-0 kafka]$ netstat -lnt | grep -Eq 'tcp6?[[:space:]]+[0-9]+[[:space:]]+[0-9]+[[:space:]]+[^ ]+:9091.*LISTEN[[:space:]]*'
[kafka@kafka-cluster-kafka-0 kafka]$ echo $?
0

[kafka@kafka-cluster-kafka-0 kafka]$ nc -z ::1 9092
[kafka@kafka-cluster-kafka-0 kafka]$ echo $?
1
` 

So, I looked at the liveness probe to understnd why it did not restarted. It seems we are checking 9091 in netstat LISTEN state, but we are not checking its actual connection. 

Any recommendations, why it may be happening and is okay to modify liveness probe to have additional checks to have kubernetes restart them based on situations like above.

**Comment by scholzj** [comment]
That is correct and it is by design. You do not want to spam the broker with connections on all the different ports (among other reasons because it would be spamming the logs with errors as well). You also cannot check any other connections because you cannot assume what connections should or should not be there.

**Comment by vshinde727** [comment]
OK, understood. In my specific case, kafka was stuck (as we see in nc response code) and liveness probe was not able to catch it, so pod was Ready but unresponsive. 
It was probably bcz of cgroupv2 dynamic jvm heap calculation, I will do more tests about this behavior with jvmOptions overrided.

Thanks a lot for your inputs!!

**Reply by scholzj** [reply]
Well, I do not remember when the support for cgroups v2 was added. But Strimzi 0.29 is by now very old. So you should keep it in mind.
